### PR TITLE
Fix/master/cleanup deprecated ffi

### PR DESCRIPTION
**Pull request by jmitchell**
See #2779 for context.

### CODE REVIEW

File: test/tutorial007/expected [hunk 1]
```diff
@@ -0,0 +1 @@
+sizeof(size_t): 8 bytes
```
Comment (Melvar) [inline test/tutorial007/expected]: This looks like a problematic assumption. Tests are supposed to work everywhere.

Comment (jmitchell) [inline test/tutorial007/expected]: Oops, thanks for catching that.

**Comment by edwinb** [comment]
The test failure here is just due to my having changed the interface for readFile to return an error if it fails. Therefore also due to us being slow to merge - sorry about that! If you can fix that then I'll merge. Thanks!

**Comment by jfdm** [comment]
I am closing this as #3070 takes this PR and addresses the changes to Idris that happened mid PR. Thanks @jmitchell for the original hard work.

**Pull request closed without merging**